File: Domotización - Interfaces/Domotización - Interfaces/VentanaManual.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domotización___Interfaces
{
    internal class VentanaManual : Ventana, IElementoDomotizado
    {
        public VentanaManual(int numVentana) : base(numVentana)
        {
        }

        public override void bloquear()
        {
            Bloqueada = true;
            Console.WriteLine($"Ventana número {NumVentana} bloqueada manualmente.");
        }

        public override void desbloquear()
        {
            Bloqueada = false;
            Console.WriteLine($"Ventana número {NumVentana} desbloqueada manualmente.");
        }

        public void Timeout()
        {
            if (!Bloqueada) {  
                desbloquear();
            } else { bloquear(); }
        }
    }
}

File: DomotizacionInterfaces/src/main/java/org/example/VentanaManual.java
package org.example;

public class VentanaManual extends Ventana{

    protected VentanaManual(int numPuerta) {
        super(numPuerta);
    }

    public void bloquear()
    {
        setBloqueada(true);
        System.out.println("Ventana número " + getNumVentana() + " bloqueada manualmente");
    }

    public void desbloquear()
    {
        setBloqueada(false);
        System.out.println("Ventana número " + getNumVentana() + " desbloqueada manualmente.");
    }


}
